### DISCUSSION TITLE
Is flux v2 ready for production?

### DISCUSSION THREAD

**Discussion by ilya-git** [fluxcd/flux2#927] [Q&A] [answered]
Flux v2 seems very interesting and I'd rather upgrade from flux v1. It seems that the features that are not implemented are not relevant for me. But what I failed to figure out from any of the documentation: is it just lacking some features, but otherwise ready for production, or is it still alpha/beta that is not recommended to run in prod?

It's probably worth including this information in some readme files both for flux v1 and v2.

**Comment by dholbach** [comment, marked as answer]
v2 is on-par feature-wise with v1 now: https://toolkit.fluxcd.io/roadmap/ - the project is in a consolidation phase before the GA. We have quite a few folks who use it in production already.

Where would you expect more information about the state of v1 vs v2 now? We'd be happy to update/extend the docs/website.

**Reply by ilya-git** [reply]
Thanks for your answer, there is a disclaimer in flux v1 github, where it says that there is v2 that is actively developed, but it's not quite clear that it is ready for production. And the page that you linked did not have 100% on everything until recently so it added up to the confusion. So if you can add on the Readme on flux v1 and flux v2 github that v2 can now be used in production (or that it will soon be there) it will be of great help to many people most likely.

Do you have some rough estimate on the GA as well since you mentioned it?

**Comment by dholbach** [comment]
We're in the in-between stage now: the code is basically all there, we'll clean things up, tidy, document and test for a couple of weeks, then do the announcement.

**Reply by ilya-git** [reply]
Thanks a lot for your help!

**Comment by stefanprodan** [comment]
Starting with flux2 v0.8, the default controllers and their APIs are considered stable and **production ready**, please see the [roadmap](https://toolkit.fluxcd.io/roadmap/) for a more details.

**Reply by ilya-git** [reply]
Thanks for the answer! I'll give it a try.
